ninshiki_yolo/detector.py

```diff
@@ -63,17 +65,21 @@ def listener_callback(self, message: MsgType):
             received_frame = cv2.imdecode(received_frame, cv2.IMREAD_UNCHANGED)
 
         if (received_frame.size != 0):
-            output_img = self.detection(received_frame)
+            self.detection(received_frame)
             self.detected_object_publisher.publish(self.detection_result)
-            self.detection_result.detected_objects.clear()
 
-            cv2.imshow(self.image_subscription.topic_name, output_img)
-            cv2.waitKey(1)
-            self.get_logger().debug("once, received image and display it")
+            if self.enable_view_detection_result:
+                postprocess_frame = self.postprocess(received_frame, self.detection_result)
+                cv2.imshow(self.image_subscription.topic_name, postprocess_frame)
+                cv2.waitKey(1)
+                self.get_logger().debug("once, received image and display it")
```

Comment:
I think you should add separate log when it received the image but not displaying it (because `self.enable_view_detection_result` was set to be false.